src/sentry/web/frontend/react_page.py

```diff
@@ -107,12 +107,17 @@ def handle_auth_required(self, request: Request, *args, **kwargs) -> HttpRespons
         # For normal users, let parent class handle (e.g. redirect to login page)
         return super().handle_auth_required(request, *args, **kwargs)
 
-    def handle(self, request: Request, organization, **kwargs) -> HttpResponse:
+    def handle(self, request: Request, organization: RpcOrganization, **kwargs) -> HttpResponse:
         if "project_id" in kwargs and request.GET.get("onboarding"):
-            project = Project.objects.filter(
-                organization=organization, slug=kwargs["project_id"]
-            ).first()
-            first_event_pending.send(project=project, user=request.user, sender=self)
+            project = project_service.get_by_slug(
+                organization_id=organization.id, slug=kwargs["project_id"]
+            )
+            if project:
+                organization_service.schedule_signal(
+                    signal=first_event_pending,
+                    organization_id=organization.id,
+                    args=dict(project=project, user_id=request.user.id if request.user else None),
```

Comment:
hmmm, I'm not a fan of serializing the entire project object here either.  What about *just* passing the project_id in instead and querying the project on the other side?  Should be possible to change the handler.

Reply:
I agree with changing the signal handler if we need to make this signal cross-region.